lib/semantics/check-do.cc

```diff
@@ -395,6 +395,20 @@ class DoContext {
     }
   }
 
+  // Step expressions can't be zero (Section 11.1.7.4.1, paragraph 1).
+  // Make this a warning since it's not a constraint violation
+  void CheckZeroStep(const parser::ScalarExpr &stepExpression) {
+    MaybeExpr stepExpr{evaluate::Fold(
+        context_.foldingContext(), AnalyzeExpr(context_, stepExpression))};
+    if (stepExpr) {
+      std::optional<std::int64_t> stepValue{evaluate::ToInt64(*stepExpr)};
+      if (stepValue && stepValue.value() == 0) {
```

Comment:
`if (auto stepValue{...}; *stepValue == 0) { ... `

Reply:
Thanks.  I'll change this.

Reply:
Is the call to ```evaluate::ToInt64(*stepExpr)``` guaranteed to return something that I can dereference?

Reply:
No; that's why it's optional.